diff --git a/GW2EI.Library/GW2EI.Services/GW2EIBuilders/JsonModels/JsonLogBuilder.cs b/GW2EI.Library/GW2EI.Services/GW2EIBuilders/JsonModels/JsonLogBuilder.cs
index 1f8fd88..57a9b9d 100644
--- a/GW2EI.Library/GW2EI.Services/GW2EIBuilders/JsonModels/JsonLogBuilder.cs
+++ b/GW2EI.Library/GW2EI.Services/GW2EIBuilders/JsonModels/JsonLogBuilder.cs
@@ -270,7 +270,14 @@ internal static class JsonLogBuilder
         jsonLog.Targets = log.LogData.Logic.Targets.Select(x => JsonNPCBuilder.BuildJsonNPC(x, log, settings, skillMap, buffMap, teamMap)).ToList();
         //
         log.UpdateProgressWithCancellationCheck("Raw Format: Building Players");
-        jsonLog.Players = log.Friendlies.Select(x => JsonPlayerBuilder.BuildJsonPlayer(x, log, settings, skillMap, buffMap, damageModMap, personalBuffs, personalDamageMods, teamMap)).ToList();
+        IEnumerable<SingleActor> friendliesToBuild = log.Friendlies;
+        string? playerFilter = Environment.GetEnvironmentVariable("EI_PLAYER_FILTER");
+        if (!string.IsNullOrWhiteSpace(playerFilter))
+        {
+            string[] accountFilter = playerFilter.Split(',', StringSplitOptions.TrimEntries | StringSplitOptions.RemoveEmptyEntries);
+            friendliesToBuild = log.Friendlies.Where(x => accountFilter.Contains(x.Account, StringComparer.OrdinalIgnoreCase));
+        }
+        jsonLog.Players = friendliesToBuild.Select(x => JsonPlayerBuilder.BuildJsonPlayer(x, log, settings, skillMap, buffMap, damageModMap, personalBuffs, personalDamageMods, teamMap)).ToList();
         //
         if (log.LogMetadata.LogErrors.Any())
         {
